tests/test_compat.py

```diff
@@ -0,0 +1,22 @@
+"""
+test that function in compati behave the same
+
+"""
+
+import pytest
+from pytest_pyodide import run_in_pyodide
+
+from micropip._compat import HttpStatusError, fetch_string_and_headers
+
+
+@run_in_pyodide
+async def _inner_test_404_raise(httpserver):
```

Comment:
```suggestion
async def _inner_test_404_raise(selenium, httpserver):
```

Reply:
Also the `httpserver` fixture isn't going to work with `@run_in_pyodide`. I'm not sure what the best way is to do this, @ryanking13 do you have any suggestions?

Reply:
You can do this way (passing only string not httpserver fixture)

```py
def test_404(httpserver):
    @run_in_pyodide
    async def _inner_test_404_raise(url):
        from ... import fetch_string_and_headers
        import pytest
        with pytest.raises(HttpStatusError):
            await fetch_string_and_headers(url, {})

    httpserver.expect_request("/404").respond_with_data(
        "Not found", status=404, content_type="text/plain"
    )
    url_404 = httpserver.url_for("/404")
    _inner_test_404_raise(url_404)
```

Reply:
Ah... let me try that, thanks for the pointers.